### PR TITLE
Assimilate notmuch release_0.25.1-949c2714

### PR DESCRIPTION
**Pull request by DamienCassou**
⚠️ Please don't merge but push the branch instead.

### CODE REVIEW

**Comment by DamienCassou** [comment]
This solves https://github.com/emacscollective/emacs.g/issues/7.

File: .gitmodules [hunk 1]
```diff
@@ -43,3 +43,12 @@
 [submodule "with-editor"]
 	path = lib/with-editor
 	url = git@github.com:magit/with-editor.git
+[submodule "lib/notmuch"]
```
Comment (tarsius) [inline .gitmodules]: Please maintain the alphabetic sorting and use just "notmuch" as name.

Comment (DamienCassou) [reply .gitmodules]: I fixed it

**Comment by tarsius** [comment]
Please push it. You should be able to do that now.

But beware, this makes you the maintainer of this recipe!

After I have made significant changes on `master`, you should merge that into the drone branch. But you don't have to do that every time I push something. Usually you should do it after I merge `master` into the various `lib/*` branches, but I think I have occasionally skipped updating the drone branches in such cases.

The recommended workflows are still up for discussion and my impression is that maintaining drone branches and "configuration sets" is still to laborious and leaves to much room for mistakes. Having a few more people experimenting would help get some fresh ideas.

**Comment by DamienCassou** [comment]
> Please push it. You should be able to do that now.

I can't. I still don't have write access even though I'm a "collaborator". Moreover, I don't want to be maintainer of emacs.g, I don't use it.

> my impression is that maintaining drone branches and "configuration sets" is still too laborious and leaves to much room for mistakes

completely agree. Beyond this, these drone branches can't be shared with people who are using borg but not emacs.g like me. I think we need a better solution if sharing is desired.

**Comment by DamienCassou** [comment]
This PR changes .gitsubmodule but doesn't add the submodule itself. Should I fix that?

**Comment by tarsius** [comment]
> This PR changes .gitsubmodule but doesn't add the submodule itself. Should I fix that?

Yes, please.

**Comment by DamienCassou** [comment]
done.

**Comment by tarsius** [comment]
And finally pushed.

**Pull request closed without merging**